openmeter/customer/adapter/customer.go

```diff
@@ -291,14 +293,42 @@ func (a *adapter) GetCustomer(ctx context.Context, input customer.GetCustomerInp
 				WithSubscription(func(query *entdb.SubscriptionQuery) {
 					applyActiveSubscriptionFilter(query, clock.Now().UTC())
 					query.WithPlan()
-				}).
-				Where(customerdb.ID(input.ID)).
-				Where(customerdb.Namespace(input.Namespace))
+				})
+
+			if input.CustomerID != nil {
+				query = query.Where(customerdb.Namespace(input.CustomerID.Namespace))
+				query = query.Where(customerdb.ID(input.CustomerID.ID))
+			} else if input.CustomerKey != nil {
+				query = query.Where(customerdb.Namespace(input.CustomerKey.Namespace))
+				query = query.Where(customerdb.Key(input.CustomerKey.Key))
+			} else if input.CustomerIDOrKey != nil {
+				query = query.Where(customerdb.Namespace(input.CustomerIDOrKey.Namespace))
```

Comment:
We should have rules for resolution order here: in case the ID and the Key matches we should always return the same entity.